### PR TITLE
Refactor core/environment&stack module - part 2

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix
- Refactoring

### Detail
- All queries moved from resolvers and services to *_repositories
- Stack renamed to StackRepository

### Relates
- #741 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/base/cdkproxy/cdk_cli_wrapper.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def update_stack_output(session, stack):
     aws = SessionHelper.remote_session(stack.accountid, stack.region)
     cfn = aws.resource('cloudformation', region_name=stack.region)
     try:
-        stack_outputs = cfn.Stack(f'{stack.name}').outputs
+        stack_outputs = cfn.StackRepository(f'{stack.name}').outputs
```
Comment (dlpzx) [inline backend/dataall/base/cdkproxy/cdk_cli_wrapper.py:66]: This is not referring to the data.all Stack class but to the aws resource cfn.Stack!

File: backend/dataall/base/cdkproxy/README.md [hunk 2]
```diff
@@ -26,11 +26,12 @@ from aws_cdk import (
 )
 from dataall.base.cdkproxy.stacks import stack
 
+
```
Comment (dlpzx) [inline backend/dataall/base/cdkproxy/README.md:29]: this is also not refering to the StackRepository but to aws_cdk.Stack.

The whole block should be:

```

from aws_cdk import (
    aws_s3 as s3,
    aws_sqs as sqs,
    core,
    Stack
)


@stack(stack="mypredefinedstack")
class MyPredefinedStack(Stack):
    def __init__(self, scope, id, **kwargs):
        super().__init__(scope, id, **kwargs)
        # constructs goes here

```

File: backend/dataall/core/environment/services/environment_service.py [hunk 3]
```diff
@@ -582,106 +566,42 @@ def list_valid_user_environments(session, data=None) -> dict:
...
-
     @staticmethod
     @ResourcePolicyService.has_resource_permission(environment_permissions.LIST_ENVIRONMENT_GROUPS)
     def paginated_user_environment_groups(session, uri, data=None) -> dict:
         return paginate(
-            query=EnvironmentService.query_user_environment_groups(session, get_context().groups, uri, data),
+            query=EnvironmentRepository.query_user_environment_groups(session, get_context().groups, uri, data),
             page=data.get('page', 1),
             page_size=data.get('pageSize', 1000),
         ).to_dict()
```
Comment (dlpzx) [inline backend/dataall/core/environment/services/environment_service.py:595]: Why not calling `query_all_environment_groups` directly? in the groupService and in the paginated_all_environment_groups?

File: backend/dataall/modules/datasets/services/dataset_service.py [hunk 4]
```diff
@@ -435,8 +435,8 @@ def _deploy_dataset_stack(dataset: Dataset):
         stack_helper.deploy_stack(dataset.environmentUri)
 
     @staticmethod
-    def _create_dataset_stack(session, dataset: Dataset) -> Stack:
-        return Stack.create_stack(
+    def _create_dataset_stack(session, dataset: Dataset) -> StackRepository:
```
Comment (dlpzx) [inline backend/dataall/modules/datasets/services/dataset_service.py:438]: This function should return a `Stack` model from `dataall/core/stacks/db/stack_models.py`

**Review by dlpzx — CHANGES_REQUESTED** [review]
There are some tiny bugs, but overall looks good

File: backend/dataall/base/cdkproxy/cdk_cli_wrapper.py (hunk 1 above)
Comment (SofiaSazonova) [reply backend/dataall/base/cdkproxy/cdk_cli_wrapper.py:66]: Oh, the naming was confusing...

**Comment by noah-paige** [comment]
Tested changes locally and created an environment - all functionality working as expected

Think just need to fix the `Stack` type hint ([in the comment above](https://github.com/data-dot-all/dataall/pull/1169#discussion_r1562601537))

**Review by dlpzx — APPROVED** [review]
looks good! Ready to merge

**Pull request merged**